bson/bsonrw/extjson_parser_test.go

```diff
@@ -148,6 +148,8 @@ func TestExtJSONParserPeekType(t *testing.T) {
 			errFs: []expectedErrorFunc{expectNoError, expectError},
 		},
 		makeInvalidTestCase("lone $scope", `{"$scope": {}}`, expectError),
+		makeInvalidTestCase("empty code with unknown extra key", `{"$code":"", "0":""}`, expectError),
+		makeInvalidTestCase("non-empty code with unknown extra key", `{"$code":"foobar", "0":""}`, expectError),
```

Comment:
do these tests actually trigger this error? I don't think either of these would cause `readValue` to return an error.

Reply:
No, they trigger the error later in the conditional, which proves that the error from `readValue` is not shadowing. If it were, this would return `nil`.

Reply:
ah, of course. I see. Thanks for clarifying that